src/Microsoft.SymbolStore.Client/WindowsSymbolSever.cs

```diff
@@ -0,0 +1,234 @@
+        public Stream Stream { get; set; }
+        public long ContentLength { get; set; }
+        public bool Compressed { get; set; }
+
+        public SymbolServerResult(Stream result, long contentLength, bool compressed)
+        {
+            Stream = result;
+            ContentLength = contentLength;
+            Compressed = compressed;
+        }
+        public SymbolServerResult(WebResponse response, bool compressed)
+        {
+            Stream = response.GetResponseStream();
+            ContentLength = response.ContentLength;
+            Compressed = compressed;
+        }
+
+        public SymbolServerResult(string fileName)
+        {
+            Stream = File.OpenRead(fileName);
+            ContentLength = new FileInfo(fileName).Length;
+            Compressed = false;
+        }
+    }
+
+    public interface ISymbolServer
+    {
+        bool PreferThisServer { get; set; }
+
+        bool IsRemoteServer { get; }
+        bool IsReachable();
+        bool IsReachable(int timeout);
+        
+        SymbolServerResult FindPEFile(string filename, int buildTimeStamp, int imageSize);
+        Task<SymbolServerResult> FindPEFileAsync(string filename, int buildTimeStamp, int imageSize);
+        SymbolServerResult FindPdb(string pdbName, Guid guid, int age);
+        Task<SymbolServerResult> FindPdbAsync(string pdbName, Guid guid, int age);
+    }
+
+    public class WindowsSymbolSever : ISymbolServer
+    {
+        private string _path = null;
+        private bool _isServer = false;
+
+        public bool PreferThisServer { get; set; }
+
+        public bool IsRemoteServer
+        {
+            get
+            {
+                return _isServer;
+            }
+        }
+
+        public WindowsSymbolSever(string path)
+        {
+            if (string.IsNullOrWhiteSpace(path))
+                throw new ArgumentNullException(nameof(path));
+
+            path = path.Trim();
+            bool isServer = false;
+            if (path.StartsWith("srv*", StringComparison.OrdinalIgnoreCase))
+            {
+                isServer = true;
+                path = path.Substring(4);
+            }
+
+            if (!isServer && path.StartsWith("http:", StringComparison.OrdinalIgnoreCase) || path.StartsWith("https:", StringComparison.OrdinalIgnoreCase))
+                isServer = true;
+
+
+            _path = path;
+            _isServer = isServer;
+        }
+
+        public bool IsReachable()
+        {
+            return IsReachable(700);
+        }
+
+        public bool IsReachable(int timeout)
+        {
+            try
+            {
+                string path = _path;
+                Uri uri = new Uri(path);
+                string host = uri.DnsSafeHost;
+
+                if (!_isServer)
+                    throw new InvalidOperationException("Cannot use IsReachable if this is not a remote server.");
+                Task<IPAddress[]> addresses = System.Net.Dns.GetHostAddressesAsync(uri.DnsSafeHost);
```

Comment:
This was meant as a more 'does this server exist' method.  In the sense that tools we (CLR) write will want to check whether symweb is available or not, but actually it might not even be appropriate to put that in the library.  It might be a better choice to simply leave it to the consumer of this library and strip out IsReachable altogether.